docs/commands.md

```diff
@@ -1,68 +1,103 @@
-* destKey - key name for destination time series
+#### Complexity
+If a compaction rule exits on a timeseries `TS.ADD` performance might be reduced, the complexity of `TS.ADD` is always O(M) when M is the amount of compactions rules or O(1).
 
+#### Notes
+You can use this command to add data to an non existing timeseries in a single command.  This is the reason why the labels and retentionsecs are optional arguments.  When specified, RedisTimeSeries will check if it needs to update the labels and or retentionSecs which introduces additional complexity.
 
-## TS.ADD - append a new value to the series
+### TS.INCRBY/TS.DECRBY - Increment the latest value
 ```sql
-TS.ADD key [labels] timestamp value
+TS.INCRBY key value [RESET time-bucket] [RETENTION retentionSecs] [LABELS field value..]
 ```
-* labels - set of key-value pairs that represent metadata labels of the key. This will be used if the module was started
-with default settings. see TODO.
-* timestamp - unix timestamp (in seconds) or `*` for automatic timestamp (using the system clock)
+or
+```sql
+TS.DECRBY key value [RESET time-bucket] [RETENTION retentionSecs] [LABELS field value..]
+```
+This command can be used as a counter/gauge that get automatic history as a time series.
+
+* key - key name for timeseries
 * value - sample numeric data value (double)
 
-### Examples
+Optional args:
+   * time-bucket - time bucket for resetting the current counter in seconds
+   * retentionSecs - max age for samples compared to current time (in seconds).
+      * Default: the global retenionsecs configuration of the database. If not set, this is 0.
+      * When set to 0, the series will not be trimmed at all
+   * labels - set of key-value pairs that represent metadata labels of the key
+
+#### Notes
+You can use this command to add data to an non existing timeseries in a single command.  This is the reason why the labels and retentionsecs are optional arguments.  When specified, RedisTimeSeries will check if it needs to update the labels and or retentionSecs which introduces additional complexity.
+
+## Aggregation / Compaction / Downsampling
+
+### TS.CREATERULE - create a compaction rule
 ```sql
-TS.ADD temperature:2:32 sensor_id=2 area_id=32 1548149180 26
-TS.ADD temperature:3:11 sensor_id=3 area_id=32 1548149180 27
-TS.ADD temperature:3:11 1548149181 30
+TS.CREATERULE sourceKey destKey AGGREGATION aggType bucketSizeSeconds
+```
+* sourceKey - key name for source time series
+* destKey - key name for destination time series
+* aggType - aggregation type one of the following: avg, sum, min, max, range, count, first, last
+* bucketSizeSeconds - time bucket for aggregation in seconds
+
+> DEST_KEY should be of a `timeseries` type, and should be created before TS.CREATERULE is called.
+
+> *Performance Notice*: if a compaction rule exits on a timeseries `TS.ADD` performance might be reduced, the complexity of `TS.ADD` is always O(M) when M is the amount of compactions rules or O(1).
```

Comment:
This notice should be kept in TS.ADD IMO.

Reply:
removed it here